studio/pages/project/[ref]/auth/policies.tsx

```diff
@@ -116,6 +116,8 @@ const AuthPoliciesTables = observer(() => {
   const { ui, meta } = useStore()
   const PageState: any = useContext(PageContext)
 
+  const roles = meta.roles.list()
```

Comment:
do we need to load meta.roles first?

Reply:
yup i've loaded meta.roles in AuthLayout
although i'm now thinking that all the store loading should happen on the page level cause not all page require the same things to be loaded

i'll probbaly do that refactor in another PR too

Reply:
Not sure if this is relevant, but I'm using `get` in the stores rather than using functions so they act a bit more like props getting passed down to components. a very subtle change of thinking about things.

So when you have any fetch in a store, it could already pre filter some of the stuff you need a lot of time.

```tsx
// this is oversimplified but just for explanation 

get roles() {
  return this.data
}

get safeRoles() {
  return this.safeRoles
}

async fetchData() {
  const response = await fetch(url)
  this.roles = response.roles
  this.safeRoles = filterSafeRolesUtil(response.roles)
}

async update() {
  const response = await patch(url)
  await fetchData()
}

```

So with this in a page / component you just need this

```tsx
const { meta } = useStore()

console.log('im always updated', meta.safeRoles)
```